rewrite-java/src/main/java/org/openrewrite/java/UseStaticImport.java

```diff
@@ -58,7 +59,7 @@ private class UseStaticImportVisitor extends JavaIsoVisitor<ExecutionContext> {
         public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
             MethodMatcher methodMatcher = new MethodMatcher(methodPattern);
             J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
-            if (methodMatcher.matches(m)) {
+            if (methodMatcher.matches(m) && method.getMethodType().hasFlags(Flag.Static)) {
```

Comment:
Would be better to place this in the next condition, where we know the method type to be non-null. Also, it should be retrieved from `m` rather than `method`, so that it also supports nested calls.

Reply:
There's an unusual test which fails when we add a limitation to static methods in this recipe https://github.com/openrewrite/rewrite/blob/e5c77d5afe17070474d7b5cf6e728600c5f776af/rewrite-java-test/src/test/java/org/openrewrite/java/UseStaticImportTest.java#L76-L78